projects/ngx-sub-form/src/lib/ngx-sub-form.component.ts

```diff
@@ -66,6 +74,63 @@ export abstract class NgxSubFormComponent<ControlInterface, FormInterface = Cont
 
   private controlKeys: (keyof FormInterface)[] = [];
 
+  // instead of having the validators defined in the utils file
+  // we define them here to have access to the form types
+
+  // `ngxSubFormValidators` should be used at a group level validation
+  // with the `getFormGroupControlOptions` hook
+  protected ngxSubFormValidators = {
+    oneOf(keysArray: (keyof FormInterface)[][]): TypedValidatorFn<FormInterface> {
+      if (!keysArray || !keysArray.length || keysArray.some(keys => !keys || keys.length < 2)) {
+        throw new OneOfValidatorRequiresMoreThanOneFieldError();
+      }
+
+      return (formGroup: TypedFormGroup<FormInterface>) => {
+        const oneOfErrors: (keyof FormInterface)[][] = keysArray.reduce(
+          (acc, keys) => {
+            if (!keys.length) {
+              return acc;
+            }
+
+            let nbNotNull = 0;
+            let cpt = 0;
+
+            while (cpt < keys.length && nbNotNull < 2) {
+              const key: keyof FormInterface = keys[cpt];
+
+              const control: AbstractControl | null = formGroup.get(key as string);
+
+              if (!control) {
+                throw new OneOfValidatorUnknownFieldError(key as string);
+              }
+
+              if (!isNullOrUndefined(control.value)) {
+                nbNotNull++;
+              }
+
+              cpt++;
+            }
+
+            if (nbNotNull !== 1) {
+              acc.push(keys);
+            }
+
+            return acc;
+          },
+          [] as (keyof FormInterface)[][],
+        );
+
+        if (oneOfErrors.length === 0) {
+          return null;
+        }
+
+        return {
+          oneOf: oneOfErrors,
+        };
```

Comment:
I might be missing some complex requirement here, but this feels like it could be simplified to the following (no idea if this compiles ;) )

```ts

const notNullKeys: Array<keyof FormInterface> = keysArray.filter(() => {
  const control: AbstractControl | null = formGroup.get(key as string);

  if (!control) {
    throw new OneOfValidatorUnknownFieldError(key as string);
  }

  return !isNullOrUndefined(control.value);
});

if (notNullKeys === 1) {
  return null;
}

return { oneOf: notNullKeys };

```

Reply:
That code doesn't handle if multiple values are defined. If there's more than 1 value, which one to pick? That should just be an error IMO